2-exact_divisor.py

```diff
@@ -0,0 +1,10 @@
+n= int(input(('Enter a number please: ')))
+list=[]
+def exact_divisor(n):
+    for i in range(1,n):
+        if n % i == 0 :
+            list.append(i)
+    list.append(n)
+    print(list)
+
+exact_divisor(n)
```

Comment:
- kodlarınız çalışıyor tebrikler.
- açıklama eklerseniz güzel olur.

asağıda degisik bir cözüm var. ama bu fonksiyonu anlamanız için yield kullanımını internetten incelemeniz gerekiyor. yield çok kullanılmasada bilgi sahibi olmakta fayda var.
ama özellikle bu sorunun çözümünde değer olarak 1.000.000.000.000.000 gibi büyük bir sayı girdiğinizde aşağıdaki kodlar ile sizin çözümünüzdeki kodların şaşırtıcı derecede çözüm süresi farkına şahit olursunuz. çözüm süresinin kısalması yield ile fazla ilgili değil çözüm yönteminden kaynaklanıyor.

import math
def divisorGenerator(n):
large_divisors = []
for i in range(1, int(math.sqrt(n) + 1)):
if n % i == 0:
yield i
if i*i != n:
large_divisors.append(n / i)
for divisor in reversed(large_divisors):
yield divisor

print (list(divisorGenerator(100000000000000)))